src/OpenSpaceToolkit/Astrodynamics/Flight/System/SatelliteDynamics.cpp

```diff
@@ -0,0 +1,758 @@
+
+        case SatelliteDynamics::GravitationalPerturbationType::EGM2008:
+            return "EGM2008" ;
+
+        default:
+            throw ostk::core::error::runtime::Wrong("Gravitational Perturbation type") ;
+
+    }
+
+    return String::Empty() ;
+
+}
+
+String                          SatelliteDynamics::StringFromAtmosphericPerturbationType                        (   const   SatelliteDynamics::AtmosphericPerturbationType&  anAtmosphericPerturbationType      )
+{
+
+    switch (anAtmosphericPerturbationType)
+    {
+
+        case SatelliteDynamics::AtmosphericPerturbationType::None:
+            return "None" ;
+
+        case SatelliteDynamics::AtmosphericPerturbationType::Exponential:
+            return "Exponential" ;
+
+        case SatelliteDynamics::AtmosphericPerturbationType::USSA1976:
+            return "USSA1976" ;
+
+        case SatelliteDynamics::AtmosphericPerturbationType::JacchiaRoberts:
+            return "JacchiaRoberts" ;
+
+        case SatelliteDynamics::AtmosphericPerturbationType::NRLMISIS00:
+            return "NRLMISIS00" ;
+
+        default:
+            throw ostk::core::error::runtime::Wrong("Atmospheric Perturbation type") ;
+
+    }
+
+    return String::Empty() ;
+
+}
+
+String                          SatelliteDynamics::StringFromThirdBodyPerturbationType                          (   const   SatelliteDynamics::ThirdBodyPerturbationType&  aThirdBodyPerturbationType  )
+{
+
+    switch (aThirdBodyPerturbationType)
+    {
+
+        case SatelliteDynamics::ThirdBodyPerturbationType::None:
+            return "None" ;
+
+        case SatelliteDynamics::ThirdBodyPerturbationType::Luni:
+            return "Luni" ;
+
+        case SatelliteDynamics::ThirdBodyPerturbationType::Solar:
+            return "Solar" ;
+
+        case SatelliteDynamics::ThirdBodyPerturbationType::LuniSolar:
+            return "LuniSolar" ;
+
+        default:
+            throw ostk::core::error::runtime::Wrong("Third Body Perturbation type") ;
+
+    }
+
+    return String::Empty() ;
+
+}
+
+String                          SatelliteDynamics::StringFromStateVectorType                                    (   const   SatelliteDynamics::StateVectorType&  aStateVectorType      )
+{
+
+    switch (aStateVectorType)
+    {
+
+        case SatelliteDynamics::StateVectorType::PositionVelocity:
+            return "PositionVelocity" ;
+
+        case SatelliteDynamics::StateVectorType::PositionVelocitywithDragCoefficient:
+            return "PositionVelocitywithDragCoefficient" ;
+
+        default:
+            throw ostk::core::error::runtime::Wrong("State Vector type") ;
+
+    }
+
+    return String::Empty() ;
+
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void                            SatelliteDynamics::TwoBody_Dynamics         (   const   SatelliteDynamics::StateVector&     x,
```

Comment:
Use global dynamical function with `FThirdBody`, `FGravitational`, `FSRP` etc appearing in general diff equation after perturbations enums are moved and set in `Environment` directly